vortex-array/src/arrays/primitive/vtable/operator.rs

```diff
@@ -61,3 +71,117 @@ impl OperatorVTable<PrimitiveVTable> for PrimitiveVTable {
         Ok(None)
     }
 }
+
+impl PipelinedSource for PrimitiveArray {
+    fn bind_source(&self, ctx: &mut dyn BindContext) -> VortexResult<Box<dyn SourceKernel>> {
```

Comment:
It's a lot of code for just a primitive array!

I imagine when we actually run in a pipelined fashion, for canonical inputs we just pass a zero-copy view of them into the pipeline. vs here where we copy the elements of the array into the mutable output.